client/src/app/primitives/__tests__/ModalSpec.js

```diff
@@ -53,6 +53,25 @@ describe('<Modal>', function() {
   });
 
 
+  describe('hideCloseIcon parameter', function() {
+
+    it('should render close icon if hideCloseIcon non existent', function() {
```

Comment:
Note that we should also disable the close on escape and close on backdrop functionality if we hide the close icon.

Three ways exists how the user can escape normal modals:

* press escape
* click backdrop (grey overlay)
* press close button

Given that it is not only about hiding the close button, won't be "noUserEscape" or "confirm" (=> user must accept or reject using modal provided buttons) a better name for the prop?
We should probably extract that new infrastructure / Modal feature into its own commit.

Reply:
This feature also disables escape/backgroup click related modal escapes so I think you're right about the naming. "noUserEscape" seems like a good naming for this.

The noUserEscape feature is already in a separated commit: https://github.com/camunda/camunda-modeler/pull/1597/commits/c27d749c449233e8c82f42f232f9d2cf70f6a237

Did you mean something else?

Reply:
I actually now realized that ESC/Background click does not work because we don't pass an onClose method to the modal, so I think naming is correct here. We only want to hide the escape icon visually.

Reply:
Why can't we be consistent then => no `onClose` method => no `X` being shown?